pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -75,48 +85,44 @@ func (cs *ClusterStore) GetPilotClusters() []*k8s_cr.Cluster {
 }
 
 // ReadClusters reads multiple clusters from a ConfigMap
-func ReadClusters(k8s kubernetes.Interface, configMapName string, configMapNamespace string) (*ClusterStore, error) {
+func ReadClusters(k8s kubernetes.Interface, configMapName string,
+	configMapNamespace string, cs *ClusterStore) error {
 
 	// getClustersConfigs
-	cs, err := getClustersConfigs(k8s, configMapName, configMapNamespace)
-	if err != nil {
-		return nil, err
+	if err := getClustersConfigs(k8s, configMapName, configMapNamespace, cs); err != nil {
+		return err
 	}
 	if len(cs.clusters) == 0 {
-		return nil, fmt.Errorf("no kubeconf found in provided ConfigMap: %s/%s", configMapNamespace, configMapName)
+		return fmt.Errorf("no kubeconf found in provided ConfigMap: %s/%s", configMapNamespace, configMapName)
 	}
 
-	return cs, nil
+	return nil
 }
 
 // getClustersConfigs(configMapName,configMapNamespace)
-func getClustersConfigs(k8s kubernetes.Interface, configMapName, configMapNamespace string) (*ClusterStore, error) {
-	cs := &ClusterStore{
-		clusters:      []*k8s_cr.Cluster{},
-		clientConfigs: map[string]clientcmdapi.Config{},
-	}
+func getClustersConfigs(k8s kubernetes.Interface, configMapName, configMapNamespace string, cs *ClusterStore) error {
 
 	clusterRegistry, err := k8s.CoreV1().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
 	if err != nil {
-		return &ClusterStore{}, err
+		return err
 	}
 
 	for key, data := range clusterRegistry.Data {
 		cluster := k8s_cr.Cluster{}
 		decoder := yaml.NewYAMLOrJSONDecoder(strings.NewReader(data), 4096)
 		if err = decoder.Decode(&cluster); err != nil {
 			log.Errorf("failed to decode cluster definition for: %s error: %v", key, err)
-			return &ClusterStore{}, err
+			continue
```

Comment:
Nice fix - I think the old code was failing if any cluster was bad.

Reply:
+1